python/cudf/cudf/core/multiindex.py

```diff
@@ -392,13 +393,124 @@ def _popn(self, n):
         return result
 
     def __repr__(self):
-        return (
-            "MultiIndex(levels="
-            + str(self.levels)
-            + ",\ncodes="
-            + str(self.codes)
-            + ")"
+        max_seq_items = get_option("display.max_seq_items") or len(self)
+
+        if len(self) > max_seq_items:
+            n = int(max_seq_items / 2) + 1
+            indices = list(range(n))
+            indices.extend(list(range(len(self) - n, len(self))))
```

Comment:
If `n` is large here this will be very expensive. Can we generate `indices` on the GPU somehow?

Reply:
Moved the indices generation to GPU